src/commands.ts

```diff
@@ -201,6 +206,104 @@ export class CosmosDBCommands {
             return "The name cannot end in a period."
         }
     }
+
+    public static async createDocDBDatabase(server: CosmosDBResourceNode, explorer: CosmosDBExplorer) {
+        const databaseName = await vscode.window.showInputBox({
+            placeHolder: 'Database Name',
+            validateInput: CosmosDBCommands.validateDatabaseName,
+            ignoreFocusOut: true
+        });
+        if (databaseName) {
+            const masterKey = await server.getPrimaryMasterKey();
+            const endpoint = await server.getEndpoint();
+            let client = new DocumentClient(endpoint, { masterKey: masterKey });
+            client.createDatabase({ id: databaseName }, async function (err, created) {
+                if (err) {
+                    vscode.window.showErrorMessage(err.body);
```

Comment:
As discussed offline - don't ever use showErrorMessage. Let the error bubble up so that telemetry catches it

(I thought VS Code always showed messages, but they don't. So you will have to modify the code here to show a message instead of throwing the error:
https://github.com/Microsoft/vscode-cosmosdb/blob/master/src/extension.ts#L91 )